pkg/flowaggregator/flowaggregator.go

```diff
@@ -213,40 +234,125 @@ type flowAggregator struct {
 		registry:                    registry,
 		set:                         ipfixentities.NewSet(false),
-		flowAggregatorAddress:       flowAggregatorAddress,
-		includePodLabels:            includePodLabels,
+		flowAggregatorAddress:       opts.FlowAggregatorAddress,
+		includePodLabels:            opts.IncludePodLabels,
 		k8sClient:                   k8sClient,
 		observationDomainID:         observationDomainID,
 		podInformer:                 podInformer,
 		sendJSONRecord:              sendJSONRecord,
+		updateCh:                    make(chan updateMsg),
+		configFile:                  configFile,
+		configWatcher:               opts.ConfigWatcher,
+		configData:                  opts.CurrentConfiguration,
+		APIServer:                   opts.Config.APIServer,
+	}
+	err := fa.InitCollectingProcess()
+	if err != nil {
+		return nil, fmt.Errorf("error when creating collecting process: %v", err)
+	}
+	err = fa.InitAggregationProcess()
+	if err != nil {
+		return nil, fmt.Errorf("error when creating aggregation process: %v", err)
+	}
+	if opts.Config.ClickHouse.Enable {
+		chInput := clickhouseclient.ClickHouseInput{
+			Username:       os.Getenv("CH_USERNAME"),
+			Password:       os.Getenv("CH_PASSWORD"),
+			Database:       opts.Config.ClickHouse.Database,
+			DatabaseURL:    opts.Config.ClickHouse.DatabaseURL,
+			Debug:          opts.Config.ClickHouse.Debug,
+			Compress:       opts.Config.ClickHouse.Compress,
+			CommitInterval: opts.ClickHouseCommitInterval,
+		}
+		err = fa.InitDBExportProcess(chInput)
+		if err != nil {
+			return nil, fmt.Errorf("error when creating db export process: %v", err)
+		}
 	}
 	podInformer.Informer().AddIndexers(cache.Indexers{podInfoIndex: podInfoIndexFunc})
-	return fa
+	return fa, nil
+}
+
+// genObservationDomainID generates an IPFIX Observation Domain ID when one is not provided by the
+// user through the flow aggregator configuration. It will first try to generate one
+// deterministically based on the cluster UUID (if available, with a timeout of 10s). Otherwise, it
+// will generate a random one. The cluster UUID should be available if Antrea is deployed to the
+// cluster ahead of the flow aggregator, which is the expectation since when deploying flow
+// aggregator as a Pod, networking needs to be configured by the CNI plugin.
+func genObservationDomainID(k8sClient kubernetes.Interface) uint32 {
+	const retryInterval = time.Second
+	const timeout = 10 * time.Second
+	const defaultAntreaNamespace = "kube-system"
+
+	clusterIdentityProvider := clusteridentity.NewClusterIdentityProvider(
+		defaultAntreaNamespace,
+		clusteridentity.DefaultClusterIdentityConfigMapName,
+		k8sClient,
+	)
+	var clusterUUID uuid.UUID
+	if err := wait.PollImmediate(retryInterval, timeout, func() (bool, error) {
+		clusterIdentity, _, err := clusterIdentityProvider.Get()
+		if err != nil {
+			return false, nil
+		}
+		clusterUUID = clusterIdentity.UUID
+		return true, nil
+	}); err != nil {
+		klog.Warningf(
```

Comment:
not your code, but please replace this with `klog.InfoS`

Reply:
Fixed.